crates/cast/bin/cmd/create2.rs

```diff
@@ -136,36 +136,49 @@ impl Create2Args {
         let mut handles = Vec::with_capacity(n_threads);
         let found = Arc::new(AtomicBool::new(false));
         let timer = Instant::now();
+
+        // Loops through all possible salts in parallel until a result is found.
+        // Each thread iterates over `(i..).step_by(n_threads)`.
         for i in 0..n_threads {
+            // Create local copies for the thread.
             let increment = n_threads;
-            let start = i;
             let regex = regex.clone();
             let regex_len = regex.patterns().len();
             let found = Arc::clone(&found);
             handles.push(std::thread::spawn(move || {
+                // Read the first bytes of the salt as a usize to be able to increment it.
                 struct B256Aligned(B256, [usize; 0]);
                 let mut salt = B256Aligned(B256::ZERO, []);
+                // SAFETY: B256 is aligned to `usize`.
                 let salt_word = unsafe { &mut *salt.0.as_mut_ptr().cast::<usize>() };
-                *salt_word = start;
+
+                // Important: set the salt to the start value, otherwise all threads loop over the
```

Comment:
the loop only covers u64 or u32 depending on the arch, so not all the available 32bytes?

fine?

Reply:
```
>> u32::MAX / 128
33554431
>> u64::MAX / 128
144115188075855871
```

iterations needed to overflow, I don't think that's reachable